shared/test/test_email_validator.rb

```diff
@@ -0,0 +1,52 @@
+# coding: utf-8
+require 'minitest/autorun'
+require 'rack/test'
+
+require_relative '../../lib/cdo/email_validator'
+
+class EmailValidatorTest < Minitest::Unit::TestCase
+
+  def test_valid_addresses
+    valid_addresses = [
+      'email@example.com', 
+      'firstname.lastname@example.com', 
+      'email@subdomain.example.co.uk', 
+      'firstname+lastname@example.com', 
+      'email@123.123.123.123', 
+      'email@[123.123.123.123]', 
+      '123456789@example.com', 
+      'email@example-one.com', 
+      '_______@example.com', 
+      'email@example.name', 
+      'email@example.museum', 
+      'email@example.co.jp', 
+      'firstname-lastname@example.com']
+
+    for address in valid_addresses 
```

Comment:
Style/For: Prefer `each` over `for`.<br>Style/TrailingWhitespace: Trailing whitespace detected.